airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -169,7 +179,11 @@ def project_id(self) -> Optional[str]:
         :return: id of the project
         :rtype: str
         """
-        return self._get_field('project')
+        overridden_project_id = self._get_field('project')
```

Comment:
I wonder if this piece of code should not be transferred to the _get_credentials_and_project_id method. Then this method will never return an invalid value and it will be easier to use it.

Reply:
Updated.